Port original nf-core training

FriederikeHanssen opened the pull request:
This PR ports the original nf-core training over. 

Changes to the original training:

- Removed warning about old Nextflow version. 24.10 is supported in codespaces/devcontainer and should work. Minimal required version for nf-demo is 24.04.2 which is out for close to 1 year now
- Invert nf-core pipeline creation: First toggle all off, then enable what we need. Aim: be more intentional. Avoid accidentally leaving in a lot of fluff. Avoid having to rewrite this every time a new things gets added by tools
- Removed further features in the nf-core pipeline creation that we did not use during the training:
    - Code linting
    -  License file
    - Enable email updates
    -  Add Seqera Platform output
    - (new but not added: RO-Crate, Render website admonitions in VSCode)
- Add hint on complex workflows in the `Check the input data` section
- Updated language, images paths

pinin4fjords reviewed (changes requested):
I think this is a tricky one, and I don't think we had time to properly refine this when it was added previously.

My overriding feeling is that it's very long, and gets very technical very quickly. 

In the short term, I think we could do with injecting some narrative, and keeping on-message:

 - There needs to be an intro where you tell people what they will be doing, and a conclusion where you iterate over the main learning outcomes, and in between narrative linking the sections. 
- Be brutal in removing detail not central to the learning outcomes. I think we should concentrate on how nf-core can help you write workflows.
- Might not be doable in current timelines, but would be great if there was a mission from the start to tie things together: "We're going to build an nf-core workflow of X modules to do Y, taking input from a sample sheet, validating it, and producing a multiqc output". Then you relate every 'Takeaway' to progression towards that objective.

**Longer-term**

I think this actually a course in itself really. We could do with building up more progressively in layers.

FriederikeHanssen commented on docs/side_quests/nf-core.md:
````diff
@@ -0,0 +1,1402 @@
+nextflow pull nf-core/demo
+```
+
+Nextflow will `pull` the pipeline's default GitHub branch.
````
do you want to remove the entire section around where the code is getting stored etc?

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1402 @@
+
+<!-- TODO Possibly leave a hint about machine types !!! hint "Changing the machine tye version"
+
+    Depending on the Nextflow version you have installed, this command might fail due to a version mismatch.
```
agreed. This was from the previous training and I removed it because it now "just" works for the version

The example requires that people actually use the 16GB machine recommended in the env setup. I suppose we can assume they used that

FriederikeHanssen replied:
fixed

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1402 @@
+
+!!!tip
+
+    Pulled pipelines are stored in a hidden assets folder. By default, this folder is `$HOME/.nextflow/assets`, but in this training environment the folder has been set to `$NXF_HOME/assets`:
```
I think this might all come from this being a separate course on running a pipeline that is currently dropped from the hello-series. I can remove this for now as it will come back at some point to the user journey

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1402 @@
+This helps to increase the reliability and reproducibility of scientific analyses and ultimately enables researchers to accelerate their scientific discoveries.
+
+nf-core is published in Nature Biotechnology: [Nat Biotechnol 38, 276–278 (2020). Nature Biotechnology](https://www.nature.com/articles/s41587-020-0439-x). An updated preprint is available at [bioRxiv](https://www.biorxiv.org/content/10.1101/2024.05.10.592912v1).
+
```
added

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1402 @@
+
+## nf-core pipelines and other components
+
+The nf-core collection currently offers [over 100 pipelines](https://nf-co.re/pipelines/) in various stages of development, [72 subworkflows](https://nf-co.re/subworkflows/) and [over 1300 modules](https://nf-co.re/modules/) that you can use to build your own pipelines.
+
+Each released pipeline has a dedicated page that includes 6 documentation sections:
+
+- **Introduction:** An introduction and overview of the pipeline
+- **Usage:** Descriptions of how to execute the pipeline
+- **Parameters:** Grouped pipeline parameters with descriptions
+- **Output:** Descriptions and examples of the expected output files
+- **Results:** Example output files generated from the full test dataset
+- **Releases & Statistics:** Pipeline version history and statistics
+
+You should read the pipeline documentation carefully to understand what a given pipeline does and how it can be configured before attempting to run it.
```
I reshuffled the sections a bit and added a dedicated: where to find pipeline information to the run part. lmk what you think about that.

FriederikeHanssen commented:
> I think this actually a course in itself really. We could do with building up more progressively in layers.

💯 https://github.com/nextflow-io/training/issues/546

FriederikeHanssen commented:
- moved everything down one section to better split it into running and developing.
- fixed the svgs that were not rendered
- added introduction
- added more extensive take away messages
- shuffled the beginning a bit. lmk what you think
- @pinin4fjords not sure I understood you correctly about the nextflow pull thing. I adapted it a bit now.

pinin4fjords commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1427 @@
+# nf-core
+
+nf-core is a community effort to develop and maintain a curated set of analysis pipelines built using Nextflow.
```
Totally open question- do you think we could use this opportunity to stress the independence of the nf-core community? 

```suggestion
nf-core is a community effort to develop and maintain a curated set of analysis pipelines built using Nextflow. The community is coordinated like X....
```

pinin4fjords commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1427 @@
+- **A short description of your pipeline:** My first pipeline
+- **Name of the main author / authors:** < YOUR NAME >
+
+5. On the Template features screen, set "Toggle all features" to **off**, then **enable**:
```
this is rendering as part of the parent list

pinin4fjords commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1427 @@
+
+1.  `tuple val(meta), path(reads)`
+
+    - A tuple with a meta _map_ (we will talk about meta maps more in the next section) and a list of FASTQ _files_
```
```suggestion
    - A tuple (basically a fixed-length list) with a meta _map_ (we will talk about meta maps more in the next section) and a list of FASTQ _files_
```

Not sure tuple is familiar to everyone

pinin4fjords commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1427 @@
+
+#### Add a parameter to the `seqtk/trim` tool
+
+nf-core modules should be flexible and usable across many different pipelines. Therefore, tool parameters are typically not set in an nf-core/module. Instead, additional configuration options on how to run the tool, like its parameters or filename, can be applied to a module using the `conf/modules.config` file on the pipeline level. Process selectors (e.g., `withName`) are used to apply configur ...
```
```suggestion
nf-core modules should be flexible and usable across many different pipelines. Therefore, optional tool parameters are typically not set in an nf-core/module. Instead, additional configuration options on how to run the tool, like its parameters or filename, can be applied to a module using the `conf/modules.config` file on the pipeline level. Process selectors (e.g., `withName`) are used to apply configuration options to modules selectively. Process selectors must be used within the `process` scope.
```

pinin4fjords commented on docs/side_quests/nf-core.md:
````diff
@@ -0,0 +1,1427 @@
+skip_trim                   = false
+```
+
+#### Adding parameters to your pipeline
````
```suggestion
#### Adding parameters to your pipeline

We can a new parameter `skip_trim` to your `nextflow.config` file and set it to `false`.

```groovy title="nextflow.config" linenums="15"
// Trimming
skip_trim                   = false
```

```

pinin4fjords reviewed (commented):
Last batch of comments after updates

FriederikeHanssen commented on docs/side_quests/nf-core.md:
````diff
@@ -0,0 +1,1427 @@
+
+The `test` profile for `nf-core/demo` is shown below:
+
+```groovy title="conf/test.config" linenums="1"
````
I didn't know this was possible 🤯

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1427 @@
+
+We will first run a pipeline in this directory and then build our own. We need the `sequencer_samplesheet.csv` for part 2. For now you can ignore it.
+
+## Part I: Run nf-core pipelines
```
can you explain what you mean? Remove Part i/II?

pinin4fjords replied:
Yep, just have numbered sections and refer to them directly. "In section 2 we will, in section 3 we will..."

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1427 @@
+//
+// MODULE: Run SEQTK_TRIM
+//
+if (!params.skip_trim) {
```
I have the feeling something got removed here. I'll dig into the git history

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1436 @@
+    ch_trimmed  = SEQTK_TRIM.out.reads
+    ch_versions = ch_versions.mix(SEQTK_TRIM.out.versions.first())
+}
+
```
```suggestion
```

FriederikeHanssen commented on docs/side_quests/nf-core.md:
````diff
@@ -0,0 +1,1435 @@
+
+[47/fdf9de] process > MYORG_MYFIRSTPIPELINE:MYFIRSTPIPELINE:SEQTK_TRIM (SAMPLE2_PE) [100%] 1 of 1 ✔
+[2a/a742ae] process > MYORG_MYFIRSTPIPELINE:MYFIRSTPIPELINE:MULTIQC                 [100%] 1 of 1 ✔
+```
````
fixed. added also missing output titles to the console statement

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -0,0 +1,1435 @@
+    - Add the author name: `GitHub Username: (@<your-name>):`
+    - Accept the defaults for the remaining prompts by typing `enter`
+
+This will create a new file in `modules/local/fastqe.nf` that already contains the container and conda definitions, the general structure of the process, and a number of TODO statements to guide you through the adaptation.
```
fixed 
<img width="642" alt="Screenshot 2025-03-11 at 09 52 33" src="https://github.com/user-attachments/assets/aad8f45a-3ff0-4c75-860d-dd85654368d1" />

FriederikeHanssen commented on docs/side_quests/nf-core.md:
```diff
@@ -1146,17 +1146,23 @@ We can comment the `ch_samplesheet.view()` line or remove it. We are not going t
-ch_seqtk_in = ch_samplesheet.branch { meta, reads ->
-    to_trim: meta["sequencer"] == "sequencer2"
-    other: true
+if (!params.skip_trim) {
+
+    ch_seqtk_in = ch_samplesheet.branch { meta, reads ->
+        to_trim: meta["sequencer"] == "sequencer2"
+        other: true
+    }
+
+    SEQTK_TRIM (
+        ch_seqtk_in.to_trim
+    )
+    ch_trimmed  = SEQTK_TRIM.out.reads
+    ch_versions = ch_versions.mix(SEQTK_TRIM.out.versions.first())
```
ok I realised I was not consistent with the indentation. INdented it now
